processing/src/main/java/io/druid/segment/filter/AndFilter.java

```diff
@@ -71,6 +77,45 @@ public ValueMatcher makeMatcher(ValueMatcherFactory factory)
     return makeMatcher(matchers);
   }
 
+  @Override
+  public ValueMatcher makeMatcher(
+      BitmapIndexSelector selector,
+      ValueMatcherFactory valueMatcherFactory,
+      RowOffsetMatcherFactory rowOffsetMatcherFactory
+  )
+  {
+    final List<ValueMatcher> offsetMatchers = new ArrayList<>();
+    final List<ValueMatcher> valueMatchers = new ArrayList<>();
+
+    for (Filter filter : filters) {
+      ValueMatcher matcher = filter.makeMatcher(selector, valueMatcherFactory, rowOffsetMatcherFactory);
+      if (filter.supportsBitmapIndex(selector)) {
+        offsetMatchers.add(matcher);
```

Comment:
@nishantmonu51 I went ahead and implemented this, only one bitmap offsetMatcher is created for And/OrFilter now